PROTOCOL.md

```diff
@@ -60,21 +60,164 @@ Here are the list of APIs used by Delta Sharing Protocol. All of the REST APIs u
-URL | {prefix}/shares
-Query Parameters | **maxResults** (type: Int, optional): The maximum number of results per page that should be returned. If the number of available results is larger than `maxResults`, the response will provide a `nextPageToken` that can be used to get the next page of results in subsequent list requests. The server may return fewer than `maxResults` items even if there are more available. The client should check `nextPageToken` in the response to determine if there are more available.<br><br>**pageToken** (type: String, optional): Specifies a page token to use. Set `pageToken` to the `nextPageToken` returned by a previous list request to get the next page of results. `nextPageToken` will not be returned in a response if there are no more results available.
-Response Header | Content-Type: application/json; charset=utf-8
-Response Body | <pre>{<br>  "items": [<br>    {<br>      "name": string<br>    }<br>  ],<br>  "nextPageToken": string,<br>}</pre> Note: the `items` field may be an empty array or missing when no results are found. The client must handle both cases.
+Method | `GET`
+Header | `Authorization: Bearer {token}`
+URL | `{prefix}/shares`
+Query Parameters | **maxResults** (type: Int32, optional): The maximum number of results per page that should be returned. If the number of available results is larger than `maxResults`, the response will provide a `nextPageToken` that can be used to get the next page of results in subsequent list requests. The server may return fewer than `maxResults` items even if there are more available. The client should check `nextPageToken` in the response to determine if there are more available. Must be non-negative. 0 will return no results but `nextPageToken` may be populated.<br><br>**pageToken** (type: String, optional): Specifies a page token to use. Set `pageToken` to the `nextPageToken` returned by a previous list request to get the next page of results. `nextPageToken` will not be returned in a response if there are no more results available.
+
+<details open>
+<summary><b>200: The shares were successfully returned.</b></summary>
+
+<table>
+<tr>
+<th>HTTP Response</th>
+<th>Value</th>
+</tr>
+<tr>
+<td>Header</td>
+<td>
+
+`Content-Type: application/json; charset=utf-8`
+
+</td>
+</tr>
+<tr>
+<td>Body</td>
+<td>
+
+```json
+{
+  "items": [
+    {
+      "name": "string"
+    }
+  ],
+  "nextPageToken": "string"
+}
+```
 
-Example:
+Note: the `items` field may be an empty array or missing when no results are found. The client must handle both cases.
+
+Note: the `nextPageToken` field may be an empty string or missing when there are no additional results. The client must handle both cases.
+</td>
+</tr>
+</table>
+</details>
+<details>
+<summary><b>400: The request is malformed.</b></summary>
+
+<table>
+<tr>
+<th>HTTP Response</th>
+<th>Value</th>
+</tr>
+<tr>
+<td>Header</td>
+<td>Content-Type: application/json</td>
+</tr>
+<tr>
+<td>Body</td>
+<td>
+
+```json
+{
```

Comment:
Do you plan to add this response format to OSS Delta Sharing server soon? If not, it's better to remove it from the protocol for now.

Reply:
Yes, will prepare a PR to add now.

Reply:
https://github.com/delta-io/delta-sharing/pull/89

I have also changed error_code to errorCode in the protocol. I think it is theoretically possible to change the behaviour inside DB's delta sharing servers to output camel case.